app/main.go

```diff
@@ -70,8 +74,31 @@ func main() {
 		log.Fatalf("Error creating counter: %s", err)
 	}
 
-	http.HandleFunc("/", handler)
-	log.Fatal(http.ListenAndServe(":8080", nil))
+	// SIGINT handles Ctrl+C locally.
+	// SIGTERM handles Cloud Run termination signal.
+	signal.Notify(signalChan, syscall.SIGINT, syscall.SIGTERM)
+
+	// Start HTTP server.
+	srv := &http.Server{
+		Addr:    ":8080",
+		Handler: http.HandlerFunc(handler),
+	}
+	go func() {
+		http.HandleFunc("/", handler)
+		if err := srv.ListenAndServe(); err != nil && err != http.ErrServerClosed {
+			log.Fatal(err)
+		}
+	}()
+
+	// Receive output from signalChan.
+	sig := <-signalChan
+	log.Printf("%s signal caught. Graceful Shutdown.", sig)
+
+	// Gracefully shutdown the server by waiting on existing requests (except websockets).
```

Comment:
Traces and metrics aren't actually being collected in the shutdown path are they? You've added the right skeleton to handle trapping the SIGTERM and gracefully shutting down but I don't see the actual flush of telemetry on shutdown.

Most of the telemetry logic is invoked in `handler` and that is not called on the shutdown path as far as I can see.

Reply:
I think it should be handled by `defer tp.Shutdown(ctx)` and `defer provider.Shutdown(ctx)` statements above.  Might be good to double-check, though.

Reply:
Yes, as @dashpole said, my understanding of using the otel sdk is that `defer tp.Shutdown(ctx)` and `defer provider.Shutdown(ctx)` handled the graceful shutdown of otel.

Reply:
Ah gotcha, thanks!